source/switch/ui.cpp

```diff
@@ -338,6 +354,33 @@ UIResult uiLoop(u32 keysDown) {
 				generateSettingString(setting);
 
 				return resultNone;
+			} else if (state == stateCheats) {
+				if ((keysDown & KEY_B)) return resultCloseCheats;
+
+				if (cursor < cheatsNumber) {
+					if(cheatsList[cursor].enabled)
+						cheatsDisable(cursor);
+					else
+						cheatsEnable(cursor);
+				} else {
+					std::string cheatCode = hbkbd::keyboard("Enter the gameshark cheat");
+					cheatCode.erase(remove_if(cheatCode.begin(), cheatCode.end(), isspace), cheatCode.end());
+					std::transform(cheatCode.begin(), cheatCode.end(),cheatCode.begin(), ::toupper);
+
+					if(cheatCode.length() == 16) {
+						// Gameshark v3
+						cheatsAddGSACode(cheatCode.c_str(), hbkbd::keyboard("Enter a description for the cheat").c_str(), true);
+					} else if(cheatCode.length() == 12) {
+						// Gameshark v1/2
+						cheatCode.insert(8, "0000");
+						cheatsAddGSACode(cheatCode.c_str(), hbkbd::keyboard("Enter a description for the cheat").c_str(), false);
+					} else {
+						uiStatusMsg("Invalid cheat-code. Are you sure this is a gameshark-code?");
+					}
+
+					std::cout << cheatCode << std::endl;
```

Comment:
please use printf instead(to keep the style of C++ flavored C) and add some context to the status message

Reply:
Oh, that was left over debug-code. I'll add a proper printf-statusmessage.